src/syntax-tree/ast.ts

```diff
@@ -463,6 +463,7 @@ export abstract class Statement implements CodeConstruct {
  * A statement that returns a value such as: binary operators, unary operators, function calls that return a value, literal values, and variables.
  */
 export abstract class Expression extends Statement implements CodeConstruct {
+    rootNode: Statement = null;
```

Comment:
Is the `rootNode` of an Expression always of type `Statement`? I am thinking about nested expressions. For example, in `(123 + (124 + 125))` isn't the root node of (124 + 125) the BinaryOperatorExpr (123 + ---)?

Reply:
I fixed this for you. However, as Expressions are Statements as well, therefore this didn't cause a problem.

Reply:
That's true. Forgot that one inherits from the other.